### PR TITLE
Validate plan hooks.

### PR DESCRIPTION
**Pull request by jmcarp**
[Fixes https://github.com/concourse/concourse/issues/1937]

cc @pivotal-jwinters

### CODE REVIEW

File: validate_test.go [hunk 1]
```diff
@@ -808,6 +808,36 @@ var _ = Describe("ValidateConfig", func() {
...
+			Context("when a job ensure hook has refers to a resource that does not exist", func() {
+				BeforeEach(func() {
+					job.Ensure = &PlanConfig{
+						Get: "some-nonexistent-resource",
+					}
+
+					config.Jobs = append(config.Jobs, job)
+				})
+
+				It("does not return an error", func() {
```
Comment (vito) [inline validate_test.go]: should be 'returns an error'

File: validate_test.go [hunk 2]
```diff
@@ -808,6 +808,36 @@ var _ = Describe("ValidateConfig", func() {
...
+					Expect(errorMessages).To(HaveLen(0))
+				})
+			})
+
+			Context("when a job ensure hook has refers to a resource that does not exist", func() {
```
Comment (vito) [inline validate_test.go]: 'has refers to a'

File: validate_test.go [hunk 3]
```diff
@@ -808,6 +808,36 @@ var _ = Describe("ValidateConfig", func() {
 				})
 			})
 
+			Context("when a job ensure hook has refers to a resource that does exist", func() {
```
Comment (vito) [inline validate_test.go]: 'has refers to a'

**Comment by jmcarp** [comment]
Updated.

**Comment by vito** [comment]
thanks!

**Pull request merged**